### PR TITLE
Sync `v2-3-stable` with `v2-3-test` to release `2.3.3`

### PR DESCRIPTION
**Pull request by ephraimbuddy**
Time for `2.3.3rc1`!

### CODE REVIEW

File: RELEASE_NOTES.rst [hunk 1]
```diff
@@ -21,7 +21,121 @@
...
+
+Airflow 2.3.3 (2022-07-04)
+--------------------------
+
+Significant Changes
+^^^^^^^^^^^^^^^^^^^
+
+- We've upgraded Flask Application Builder to a major version 4.*.
+
+  Flask Application Builder is one of the important components of Airflow Webserver, as
```
Comment (jedcunningham) [inline RELEASE_NOTES.rst]:
```suggestion
- We've upgraded Flask App Builder to a major version 4.*.

  Flask App Builder is one of the important components of Airflow Webserver, as
```

They use "App", not "Application".

File: RELEASE_NOTES.rst [hunk 2]
```diff
@@ -21,7 +21,121 @@
...
+  it uses a lof of dependencies that are essential to run the webserver and integrate it
+  in enterprise environments - especially authentication.
+
+  The FAB 4.* upgrades a number of dependencies to major releases, which upgrades them to versions
+  that have a number of security issues fixed. A lot of tests were performed to bring the dependencies
+  in a backwards-compatible way, however the dependencies themselves implement breaking changes in their
+  internals so it might be that some of those changes might impact the users in case they are using the
+  libraries for their onw purposes.
```
Comment (jedcunningham) [inline RELEASE_NOTES.rst]:
```suggestion
  libraries for their own purposes.
```

File: RELEASE_NOTES.rst [hunk 3]
```diff
@@ -21,7 +21,121 @@
...
+
+  * Jinja2 2.X to 3.X `breaking changes: <https://jinja.palletsprojects.com/en/3.0.x/changes/#version-3-0-0>`_
+
+  * Werkzeug 1.X to 2.X `breaking changes <https://werkzeug.palletsprojects.com/en/2.0.x/changes/#version-2-0-0>`_
+
+  * pyJWT 1.X to 2.X `breaking changes: <https://pyjwt.readthedocs.io/en/stable/changelog.html#v2-0-0>`_
+
+  * Click 7.X to 8.X `breaking changes: <https://click.palletsprojects.com/en/8.0.x/changes/#version-8-0-0>`_
+
+  * itsdangerous 1.X to 2.X `breaking changes <https://github.com/pallets/itsdangerous/blob/main/CHANGES.rst#version-200>`_ (#24399)
```
Comment (jedcunningham) [inline RELEASE_NOTES.rst]:
```suggestion
  * itsdangerous 1.X to 2.X `breaking changes <https://github.com/pallets/itsdangerous/blob/main/CHANGES.rst#version-200>`_
```

Let's move this to the first line.

File: RELEASE_NOTES.rst [hunk 4]
```diff
@@ -21,7 +21,121 @@
...
 
-Airflow 2.3.2 (2021-06-04)
+
+Airflow 2.3.3 (2022-07-04)
+--------------------------
+
+Significant Changes
+^^^^^^^^^^^^^^^^^^^
+
+- We've upgraded Flask Application Builder to a major version 4.*.
```
Comment (jedcunningham) [inline RELEASE_NOTES.rst]:
```suggestion
- We've upgraded Flask Application Builder to a major version 4.*. (#24399)
```

File: RELEASE_NOTES.rst [hunk 5]
```diff
@@ -21,7 +21,121 @@
...
+  * Click 7.X to 8.X `breaking changes: <https://click.palletsprojects.com/en/8.0.x/changes/#version-8-0-0>`_
+
+  * itsdangerous 1.X to 2.X `breaking changes <https://github.com/pallets/itsdangerous/blob/main/CHANGES.rst#version-200>`_ (#24399)
+
+
+Bug Fixes
+^^^^^^^^^
+
+- Allow for ``LOGGING_LEVEL=DEBUG`` (#23360)
+- Update Dockerfile (#24397)
```
Comment (jedcunningham) [inline RELEASE_NOTES.rst]:
```suggestion
```

File: RELEASE_NOTES.rst [hunk 6]
```diff
@@ -21,7 +21,121 @@
...
+- Parse error for task added to multiple groups (#23071)
+
+
+Doc only changes
+^^^^^^^^^^^^^^^^
+
+- docs: sqlalchemy link update (#24627)
+- Update ui.rst (#24514)
+- Update production-deployment.rst (#24121)
+- Fix inverted section levels in best-practices.rst (#23968)
```
Comment (jedcunningham) [inline RELEASE_NOTES.rst]:
```suggestion
```

I think we can skip all of these.

File: RELEASE_NOTES.rst [hunk 7]
```diff
@@ -21,7 +21,121 @@
...
+- Alphabetizes two tables (#23923)
+- Clarify that users should not use Maria DB (#24556)
+- Add imports to deferring code samples (#24544)
+- Add note about image regeneration in June 2022 (#24524)
+- Small cleanup of ``get_current_context()`` chapter (#24482)
+- Docs: Fix default 2.2.5 log_id_template (#24455)
```
Comment (jedcunningham) [inline RELEASE_NOTES.rst]:
```suggestion
- Fix default 2.2.5 log_id_template (#24455)
```

File: RELEASE_NOTES.rst [hunk 8]
```diff
@@ -21,7 +21,121 @@
...
+- Update description of installing providers separately from core (#24454)
+- Mention context variables and logging (#24304)
+
+
+Misc/Internal
+^^^^^^^^^^^^^
+
+- Remove internet explorer support (#24495)
+- Fix flaky order of returned dag runs (#24405)
```
Comment (jedcunningham) [inline RELEASE_NOTES.rst]: I think this one might be categorized as a bug?

**Review by jedcunningham — COMMENTED** [review]
Also missing the delete for the newsfragment.

File: RELEASE_NOTES.rst [hunk 9]
```diff
@@ -21,7 +21,121 @@
...
+Bug Fixes
+^^^^^^^^^
+
+- Allow for ``LOGGING_LEVEL=DEBUG`` (#23360)
+- Update Dockerfile (#24397)
+- Fix grid date ticks (#24738)
+- Debounce status highlighting in Grid view (#24710)
+- Fix Grid vertical scrolling (#24684)
+- don't try to render child rows for closed groups (#24637)
+- Restore tooltip animation (#24503)
```
Comment (bbovenzi) [inline RELEASE_NOTES.rst]:
```suggestion
- Restore tooltip animation (#24503)
```

This was fixing another commit here. Not a bug fix for release.

File: RELEASE_NOTES.rst [hunk 10]
```diff
@@ -21,7 +21,121 @@
...
+- Don't crash scheduler if exec config has old k8s objects (#24117)
+- ``TI.log_url`` fix for ``map_index`` (#24335)
+- Fix migration ``0080_2_0_2`` - Replace null values before setting column not null (#24585)
+- Patch ``sql_alchemy_conn`` if old Postgres schemes used (#24569)
+- Seed ``log_template`` table (#24511)
+- Fix deprecated ``log_id_template`` value (#24506)
+- Fix toast messages (#24505)
+- Add indexes for CASCADE deletes for task_instance (#24488)
+- Return empty dict if Pod JSON encoding fails (#24478)
+- Remove ``framer-motion`` from custom tooltip (#24449)
```
Comment (bbovenzi) [inline RELEASE_NOTES.rst]:
```suggestion
- Remove ``framer-motion`` from custom tooltip (#24449)
```
We created the custom tooltip in 2.3.3 so we probably don't need to highlight this.

File: RELEASE_NOTES.rst [hunk 11]
```diff
@@ -21,7 +21,121 @@
 
 .. towncrier release notes start
 
-Airflow 2.3.2 (2021-06-04)
+
+Airflow 2.3.3 (2022-07-04)
+--------------------------
+
+Significant Changes
```
Comment (bbovenzi) [inline RELEASE_NOTES.rst:28]: Should upgrading webpack or adding typescript to the UI be mentioned in the release notes?

Comment (ephraimbuddy) [reply RELEASE_NOTES.rst:28]: I think the mention at `Misc/Internal` is enough unless there are extra actions users must take because of the change

**Comment by ephraimbuddy** [comment]
> Also missing the delete for the newsfragment.

The newsfragment is deleted here: https://github.com/apache/airflow/pull/24762/commits/6e6ce517d3461e8cce0c4e4753a061c68dbd5d81, I wonder why it's not showing up in file changes

**Comment by eladkal** [comment]
Do we want https://github.com/apache/airflow/pull/22658 included in 2.3.3 as well?

**Comment by ephraimbuddy** [comment]
> Do we want #22658 included in 2.3.3 as well?

This is a core change and I feel we should exclude it for not being merged before now. It's late IMO

File: RELEASE_NOTES.rst [hunk 12]
```diff
@@ -21,7 +21,126 @@
...
+^^^^^^^^^^^^^^^^^^^
+
+- We've upgraded Flask App Builder to a major version 4.*. (#24399)
```
Comment (jedcunningham) [inline RELEASE_NOTES.rst]:
```suggestion
We've upgraded Flask App Builder to a major version 4.* (#24399)
""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""
```

Sorry, just noticed this wasn't converted to a heading yet.

File: RELEASE_NOTES.rst [hunk 13]
```diff
@@ -21,7 +21,126 @@
...
+
+    * ``Jinja2`` 2.X to 3.X `breaking changes: <https://jinja.palletsprojects.com/en/3.0.x/changes/#version-3-0-0>`__
+
+    * ``Werkzeug`` 1.X to 2.X `breaking changes <https://werkzeug.palletsprojects.com/en/2.0.x/changes/#version-2-0-0>`__
+
+    * ``pyJWT`` 1.X to 2.X `breaking changes: <https://pyjwt.readthedocs.io/en/stable/changelog.html#v2-0-0>`__
+
+    * ``Click`` 7.X to 8.X `breaking changes: <https://click.palletsprojects.com/en/8.0.x/changes/#version-8-0-0>`__
+
+    * ``itsdangerous`` 1.X to 2.X `breaking changes <https://github.com/pallets/itsdangerous/blob/main/CHANGES.rst#version-200>`__ (#24399)
```
Comment (jedcunningham) [inline RELEASE_NOTES.rst]:
```suggestion
    * ``itsdangerous`` 1.X to 2.X `breaking changes <https://github.com/pallets/itsdangerous/blob/main/CHANGES.rst#version-200>`__
```

File: RELEASE_NOTES.rst [hunk 14]
```diff
@@ -21,7 +21,126 @@
...
+
```
Comment (jedcunningham) [inline RELEASE_NOTES.rst:61]:
```suggestion
```

Extra newline

File: RELEASE_NOTES.rst [hunk 15]
```diff
@@ -21,7 +21,126 @@
...
+- Add TaskInstance State ``REMOVED`` to finished states and success states (#23797)
+- Fix mapped task immutability after clear (#23667)
+- Fix permission issue for dag that has dot in name (#23510)
+- Fix closing connection ``dbapi.get_pandas_df`` (#23452)
+- Check bag DAG ``schedule_interval`` match timetable (#23113)
+- Parse error for task added to multiple groups (#23071)
+- Fix flaky order of returned dag runs (#24405)
+- Migrate ``jsx`` files that affect run/task selection to ``tsx`` (#24509)
+- Fix links to sources for examples (#24386)
+
```
Comment (jedcunningham) [inline RELEASE_NOTES.rst]:
```suggestion
```

File: RELEASE_NOTES.rst [hunk 16]
```diff
@@ -21,7 +21,126 @@
...
+- Alphabetizes two tables (#23923)
+- Clarify that users should not use Maria DB (#24556)
+- Add imports to deferring code samples (#24544)
+- Add note about image regeneration in June 2022 (#24524)
+- Small cleanup of ``get_current_context()`` chapter (#24482)
+- Fix default 2.2.5 log_id_template (#24455)
+- Update description of installing providers separately from core (#24454)
+- Mention context variables and logging (#24304)
+
```
Comment (jedcunningham) [inline RELEASE_NOTES.rst:120]:
```suggestion
```

File: RELEASE_NOTES.rst [hunk 17]
```diff
@@ -21,7 +21,126 @@
...
+- The JWT claims in the request to retrieve logs have been standardized: we use ``nbf`` and ``aud`` claims for
+  maturity and audience of the requests. Also "filename" payload field is used to keep log name. (#24519)
+- Address all ``yarn`` test warnings (#24722)
+- Add tests for the ``grid_data`` endpoint (#24375)
+- Remove ``framer-motion`` from custom tooltip (#24449)
+- Upgrade to react 18 and chakra 2 (#24430)
+- Refactor ``DagRun.verify_integrity`` (#24114)
+- Upgrade FAB to 4.1.1 (#24399)
+- We now need at least Flask-WTF 0.15 (#24621)
+
```
Comment (jedcunningham) [inline RELEASE_NOTES.rst:140]:
```suggestion
```

File: RELEASE_NOTES.rst [hunk 18]
```diff
@@ -21,7 +21,126 @@
...
+- Switch Markdown engine to markdown-it-py (#19702)
+- Update 'rich' to latest version across the board. (#24186)
+- Get rid of TimedJSONWebSignatureSerializer (#24519)
+- Update flask-appbuilder ``authlib``/ ``oauth`` dependency (#24516)
+- Upgrade to ``webpack`` 5 (#24485)
+- Add ``typescript`` (#24337)
+- The JWT claims in the request to retrieve logs have been standardized: we use ``nbf`` and ``aud`` claims for
+  maturity and audience of the requests. Also "filename" payload field is used to keep log name. (#24519)
+- Address all ``yarn`` test warnings (#24722)
+- Add tests for the ``grid_data`` endpoint (#24375)
```
Comment (jedcunningham) [inline RELEASE_NOTES.rst]: This should probably be a bugfix and reworded, something like "Set proper Content-Type and chartset on ``grid_data`` endpoint".

File: RELEASE_NOTES.rst [hunk 19]
```diff
@@ -21,7 +21,126 @@
...
+- Remove ``framer-motion`` from custom tooltip (#24449)
```
Comment (jedcunningham) [inline RELEASE_NOTES.rst]: Why isn't this a bugfix?

Comment (ephraimbuddy) [reply RELEASE_NOTES.rst]: Good catch. I will combine it with `Improve grid rendering performance with a custom tooltip (#24417)`,

File: RELEASE_NOTES.rst (hunk 17 above)
Comment (ephraimbuddy) [reply RELEASE_NOTES.rst:140]: I think we should have this extra space since it's the last before we get to 2.3.2

**Review by jedcunningham — APPROVED** [review]

**Comment by hwooson12** [comment]
I was just wondering why the release doesn't include #24373. It seems like it is because the PR isn't labeled as `type:bug-fix`. I thought it was `bug-fix` so that I expected it's included in release `2.3.3` since I have the same problem with timezone in logging and look forward to this release. Is the change going to be classified as `improvement` and released in 2.4.x?

**Comment by jedcunningham** [comment]
@hwooson12, generally things sometimes fall through the cracks unfortunately. I don't think there was anything specific keeping it from the next release, it just got missed. If you do notice PRs that should be bugfixes in the next release and they aren't in the milestone, just speak up.

(In this specific case, it looks like a better fix is now being proposed anyways.)

**Comment by bbovenzi** [comment]
> (In this specific case, it looks like a better fix is now being proposed anyways.)

We're about to revert that commit and fix that issue properly in #24811.

**Pull request merged**